src/pages/List.js

```diff
@@ -1,30 +1,80 @@
 import dayjs from 'dayjs';
 import React, { useContext, useState } from 'react';
+import normalizeName from '../lib/normalizeName';
+import useListToken, { getCurrentToken } from '../useListToken';
 import { FirestoreCollection } from 'react-firestore';
-
 import { ListContext } from '../listContext';
-import useListToken, { getCurrentToken } from '../useListToken';
+import { Link } from 'react-router-dom';
+import AppHeader from '../components/AppHeader';
+import {
+  Container,
+  Loader,
+  Dimmer,
+  Input,
+  Accordion,
+  Checkbox,
+  Segment,
+  Icon,
+  Menu,
+  Responsive,
+  Button,
+  List as ListUI,
+  Divider,
+} from 'semantic-ui-react';
+
 import NavTabs from '../components/NavTabs';
-import Loading from '../components/Loading';
-import normalizeName from '../lib/normalizeName';
 
-import './List.css';
+const List = props => {
+  const {
+    shoppingList,
+    setShoppingList,
+    deleteItem,
+    purchaseItem,
+  } = useContext(ListContext);
 
-function isLessThan24hrs(datePurchased) {
-  return dayjs().diff(dayjs(datePurchased), 'hours') <= 24;
-}
+  const [filteredInput, setFilteredInput] = useState('');
 
-//we are checking if the last date it was purchased is less than 24hrs using isLessThan24hrs function
-function isChecked(lastDatePurchased) {
-  return !!lastDatePurchased && isLessThan24hrs(lastDatePurchased);
-}
+  const { token } = useListToken;
 
-const List = props => {
-  const [filteredInput, setFilteredInput] = useState('');
-  const { shoppingList, setShoppingList, purchaseItem } = useContext(
-    ListContext,
-  );
-  const { token } = useListToken();
+  const today = dayjs();
+
+  const [accordionState, setAccordionState] = useState(-1);
+
+  const handleClick = index => {
+    const activeIndex = accordionState;
+    const newIndex = activeIndex === index ? -1 : index;
+    setAccordionState(newIndex);
+  };
+  function setBackgroundColor(nextExpectedPurchase) {
+    const colorCode = {
+      SOON: 'green',
+      KIND_OF_SOON: 'orange',
+      NOT_SOON: 'red',
+    };
+
+    switch (nextExpectedPurchase) {
+      case 7:
```

Comment:
This works during testing because each item has the exact values. In practice, items will have values for this variable that are more like 7.342, 2.47, and 14.923 ... unless I'm misunderstanding, any value other than 7, 14, and 30 will result in the item getting the color white.

Instead, you could use a series of `if` statements, such as the following:

    if (nextExpectedPurchase <= 7) {
        // set to colorCode.SOON
    }
    else if (nextExpectedPurchase <= 14) {
        // set to colorCode.KIND_OF_SOON
    }}
    else if (nextExpectedPurchase <= 30) {
        // set to colorCode.NOT_SOON
    }}
    else {
        // set to white
    }}

Reply:
Changed here https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/67/commits/110f0bfd11d29b52ec533b634983e00fa9104150